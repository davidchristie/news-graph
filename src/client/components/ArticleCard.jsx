// import moment from 'moment'
import React, { PropTypes } from 'react'
import { Link } from 'react-router'

// function footer (publishedAt) {
//   const date = moment(publishedAt).format('MMMM Do YYYY, h:mm:ss a')
//   if (publishedAt) {
//     return (
//       <div className='card-footer'>
//         {date}
//       </div>
//     )
//   }
// }

export default class ArticleCard extends React.Component {
  render () {
    const article = this.props.article
    const url = article.url
    const title = article.title
    const description = article.description
    const image = article.thumbnail_url ? <img className='card-img-top img-fluid' src={article.thumbnail_url} alt='Card image cap' /> : null
    return (
      <div className='card'>
        {image}
        <div className='card-block'>
          <Link to={`/articles?url=${url}`}>
            <h4 className='card-title'>{title}</h4>
          </Link>
          <p className='card-text'>{description}</p>
        </div>
      </div>
    )
  }
}

ArticleCard.propTypes = {
  article: PropTypes.shape({
    // id: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired
  }).isRequired
}
