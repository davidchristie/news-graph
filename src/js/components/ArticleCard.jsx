import moment from 'moment'
import React, { PropTypes } from 'react'
import { Link } from 'react-router'

function footer (publishedAt) {
  const date = moment(publishedAt).format('MMMM Do YYYY, h:mm:ss a')
  if (publishedAt) {
    return (
      <div className="card-footer">
        {date}
      </div>
    )
  }
}

export default class ArticleCard extends React.Component {
  render () {
    const article = this.props.article
    const id = article.id
    const title = article.title
    const description = article.description
    // const url = article.url
    const urlToImage = article.urlToImage
    const publishedAt = article.publishedAt
    return (
      <div className="card">
        <img className="card-img-top img-fluid" src={urlToImage} alt="Card image cap"/>
        <div className="card-block">
          <Link to={`article/${id}`}>
            <h4 className="card-title">{title}</h4>
          </Link>
          <p className="card-text">{description}</p>
        </div>
        {footer(publishedAt)}
      </div>
    )
  }
}

ArticleCard.propTypes = {
  article: PropTypes.shape({
    url: PropTypes.string.isRequired
  }).isRequired
}
