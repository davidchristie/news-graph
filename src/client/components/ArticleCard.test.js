/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ArticleCard from './ArticleCard'
import article from '../examples/article'

test('ArticleCard renders correctly', () => {
  const wrapper = shallow(<ArticleCard article={article} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
