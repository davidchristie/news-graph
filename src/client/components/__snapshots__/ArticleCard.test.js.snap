exports[`test ArticleCard renders correctly 1`] = `
<div
  className="card">
  <img
    alt="Card image cap"
    className="card-img-top img-fluid"
    src="http://s1.reutersmedia.net/resources/r/?m=02&d=20170118&t=2&i=1169143813&w=&fh=545px&fw=&ll=&pl=&sq=&r=LYNXMPED0H0G0" />
  <div
    className="card-block">
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="article/1">
      <h4
        className="card-title">
        British tour operator evacuating 1,000 Gambia holidaymakers
      </h4>
    </Link>
    <p
      className="card-text">
      British tour operator Thomas Cook started evacuating nearly 1,000 holidaymakers from Gambia on Wednesday after President Yahya Jammeh declared a state of emergency on his final day in office, increasing the chances of civil unrest.
    </p>
  </div>
  <div
    className="card-footer">
    January 18th 2017, 9:29:05 pm
  </div>
</div>
`;
